[Liu Han] ip

SweetPotato0213 opened the pull request:
# Duke Helper

> “Your mind is for having ideas, not holding them.” – David Allen [Quoted from here](https://dansilvestre.com/productivity-quotes/)

### Introduction

Duke is an interactive personal chat robot designed for people tired of remembering all the tasks they've done or they're going to do. It's
- text-based
- easy to use
- retentive enough to remember all the tasks you've ever mentioned
- **free**!

### Steps to use

1. Download it from here
2. Go to the directory and double-click Duke
3. Input your tasks (including to-do, event, and deadline)
4. Let it manage your tasks for you :satisfied:

### Features

- [ ] Managing events
- [ ] Managing to-do's
- [ ] Managing dealines
- [ ] Reminders (coming soon)

### Addition

If you are a Java programmer, you can also use it to practice Java. Here's the `main` method:

```java
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

SweetPotato0213 committed:
Level 1

Greet, Echo, Exit

SweetPotato0213 committed:
Level 2. Add, List

SweetPotato0213 committed:
Level 3. Mark as Done

SweetPotato0213 committed:
Level 4. ToDos, Events, Deadlines

SweetPotato0213 committed:
A-TextUiTesting: Automated Text UI Testing

SweetPotato0213 committed:
Level 5. Handle Errors

SweetPotato0213 committed:
Level 6. Delete

SweetPotato0213 committed:
A-Enums: Use Enums

sarrrdin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,53 @@
 public class Duke {
+
+    enum Ability {
+        bye,
+        list,
+        mark,
+        unmark,
+        todo,
+        delete
```
Enum names should be written in PascalCase or all uppercase?

```
BYE,
LIST,
...
```

sarrrdin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,53 @@
+    }
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
```
I'm thinking that maybe variable logo can be set as final?

sarrrdin commented on text-ui-test/runtest.sh:
```diff
@@ -13,7 +13,7 @@ then
 
 # compile the code into the bin folder, terminates if error occurred
-if ! javac -cp ../src/main/java -Xlint:none -d ../bin ../src/main/java/*.java
+if ! javac -cp ../src/main/java -Xlint:none -d ../bin ../src/main/java/Duke.java
```
I like how you changed the accessibility where it is more specific 😄

sarrrdin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,53 @@
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        String hello = "____________________________________________________________\n" +
```
I think can leave an empty space here?

sarrrdin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,53 @@
+import java.util.*;
```
Import of classes should be listed explicitly?

sarrrdin commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,53 @@
+                "     Here are the tasks in your list:";
+
+        System.out.println(strlst);
+
```
I think can remove a blank line here 😃

sarrrdin reviewed (commented):
Looks good. 👍
Some cosmetic changes are requested to follow coding standards.
LGTM.

SweetPotato0213 committed:
Cosmetic changes to follow coding standards

Update all the codes to follow coding standards

SweetPotato0213 committed:
Level 2(v2)

correct the functions for level 2

SweetPotato0213 committed:
Level 3(v2)

Correct the functions and separate Task class from Duke class file

nmtuan2001 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+                "    ____________________________________________________________\n";
+        System.out.println(hello);
+
+        List<Task> list = new ArrayList<Task>();
```
Maybe you can change this variable name to tasks or something like that to make it clearer?

nmtuan2001 commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+
+        System.out.println(strlst);
+
+        String todolist = "____________________________________________________________\n" +
```
Should this be `todoList`?

Qiaoran-M commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+                int number = sc.nextInt();
+                Task task = list.get(number - 1);
+                task.markAsDone();
+                System.out.println("    ____________________________________________________________\n" +
```
You may consider placing line break before operator +

tensaida replied:
Also consider storing the separator as a constant

Qiaoran-M commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+
+
+
+        String strlst = "____________________________________________________________\n" +
```
Maybe you can change this variable to stringList or something like that to make it clearer?

Qiaoran-M commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+
 public class Duke {
+
+    enum Ability {
```
Good use of enums for different commands!

Qiaoran-M reviewed (commented):
LGTM!

tensaida commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+        String hello = "    ____________________________________________________________\n" +
+                "     Hello! I'm Duke\n" +
+                "     What can I do for you?\n" +
+                "    ____________________________________________________________\n";
```
Good minimalist design

tensaida commented on src/main/java/Duke.java:
```diff
@@ -1,10 +1,84 @@
+        String input = sc.next();
+
+        while (!input.equals("bye")) {
+            if (input.equals("list")) {
```
Consider using `switch` case instead of `if-else`

tensaida reviewed (commented):
You've made good progress in your iP. All the best for future weeks

SweetPotato0213 committed:
Level 4(v2)

Correct and supplement

SweetPotato0213 committed:
Test using the I/O redirection technique

SweetPotato0213 committed:
Level 5(v2)

Update and correct the codes

SweetPotato0213 committed:
Level 6(v2)

Add delete function

SweetPotato0213 committed:
Level 7.Save

Save the tasks in the hard disk automatically whenever the task list changes.

SweetPotato0213 committed:
Level 8. Dates and Times

Teach Duke how to understand dates and times. Store deadline dates as a java.time.LocalDate objects. Accept dates in yyyy-mm-dd format (e.g., 2019-10-15) and print in a different format MMM dd yyyy e.g., (Oct 15 2019).

SweetPotato0213 committed:
Level 7. Save

Do it in a new branch

SweetPotato0213 committed:
Level 8. Date time

Update code in a new branch

SweetPotato0213 committed:
Merge branch 'branch-Level-8'

SweetPotato0213 committed:
Level 7(v2)

it's for merge trial

SweetPotato0213 committed:
Merge branch 'branch-Level-7'

SweetPotato0213 committed:
Use More OOP

Added Storage, Parser, TaskList classes

SweetPotato0213 committed:
Organize classes into a package

Package all classes into duke

SweetPotato0213 committed:
Merge branch 'add-gradle-support'

SweetPotato0213 committed:
Use Gradle

Use Gradle to automate some of the build tasks of the project. Now I can run Duke.main() using Gradle but cannot run it in the terminal (If I do java Duke.java it will report 9 errors saying cannot find symbols and compilation failed. I don't know why.).

SweetPotato0213 committed:
Add JUnit Tests

create a folder (test/java/duke) to place all the test classes, use Assertions to compare the expected and actual outcomes

SweetPotato0213 committed:
Create a JAR file

Package the app as an executable JAR file so that it can be distributed easily.

SweetPotato0213 committed:
Add JavaDoc Comments

Add JavaDoc comments to the code. Add header comments to all non-private classes/methods.

SweetPotato0213 committed:
Follow the coding standard

Tweak the code to comply with a coding standard.

SweetPotato0213 committed:
Implement Find function

Give users a way to find a task by searching for a keyword.

SweetPotato0213 committed:
Merge branch 'branch-A-CodingStandard'

SweetPotato0213 committed:
Merge branch 'branch-Level-9'

SweetPotato0213 committed:
Cosmetic changes following coding standard

Place line break before operators.

SweetPotato0213 committed:
javafx test

SweetPotato0213 committed:
Added Main.java and fxml files

SweetPotato0213 committed:
Create MainWindow.java

SweetPotato0213 committed:
Added GUI

Modify parser method (return value from void to String)

SweetPotato0213 committed:
Improve Code Quality

Make some cosmetic changes to the codes according to the iP code quality feedback from teaching group

SweetPotato0213 committed:
Use Assertions

Add assert statement in printList method in Parser class

SweetPotato0213 committed:
Improve the Javadoc statement for getStatus method

SweetPotato0213 committed:
Merge pull request #2 from SweetPotato0213/branch-A-CodeQuality

Improve the Javadoc statement for getStatus method

SweetPotato0213 committed:
Merge pull request #3 from SweetPotato0213/branch-A-Assertions

Use Assertions

SweetPotato0213 committed:
Add an extension Update

SweetPotato0213 committed:
User Guide

SweetPotato0213 committed:
Attempt to get the GUI work

SweetPotato0213 committed:
CLI correction

SweetPotato0213 committed:
CLI update
